mlflow/spark.py

```diff
@@ -250,11 +253,14 @@ def _save_model_metadata(dst_dir, spark_model, mlflow_model, sample_input, conda
 
 
 def _validate_model(spark_model):
-    from pyspark.ml.pipeline import PipelineModel
-
-    if not isinstance(spark_model, PipelineModel):
-        raise MlflowException("Not a PipelineModel. SparkML can only save PipelineModels.",
-                              INVALID_PARAMETER_VALUE)
+    from pyspark.ml import Model
+    from pyspark.ml.util import MLReadable, MLWritable
+    if not isinstance(spark_model, Model) \
+            or not isinstance(spark_model, MLReadable) \
+            or not isinstance(spark_model, MLWritable):
+        raise MlflowException(
+                "Not a descendant of pyspark.Model. SparkML can only save descendants of pyspark.Model that implement MLWritable and MLReadable.", 
```

Comment:
Nit: We're always including `Not a descendant of pyspark.Model` in the error text, even if the  `isinstance(spark_model, Model)` check passes. Maybe we should say something like `Cannot serialize model input. SparkML can only save...`.

Reply:
Yeah, good point. Adjusting this and the test